File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>
using namespace std;

unordered_set<int> us;

int solution(vector<int> nums)
{
    int answer = 0;
    
    for(auto v : nums){
        us.insert(v);
    }
    
    answer = us.size();
    if(nums.size()/2 < us.size()){
        answer = nums.size()/2;
    }
    
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    
    let m = new Map();
    
    nums.forEach((v) => {
        m.set(v,m.get(v) +1);
    })
    
    if(m.size < nums.length / 2) answer = m.size
    else answer = nums.length/2
    
    return answer;
}